src/main/java/eu/bbmri/eric/csit/service/negotiator/mappers/NegotiationModelMapper.java

```diff
@@ -118,6 +127,19 @@ private String negotiationStatusConverter(String negotiationId) {
     }
   }
 
+  private Boolean negotiationPostsEnabledConverter(String negotiationId) {
+    boolean postsEnabled = false;
+    try {
+      NegotiationState currentState = negotiationLifecycleService.getCurrentState(negotiationId);
+      if (currentState.equals(NegotiationState.ONGOING)) {
```

Comment:
Please rewrite using spring state machine action, so we don't put lifecycle logic in different places through the code base, there is a guid here: https://www.baeldung.com/spring-state-machine